lib/tools/common/patching_parallel.py

```
@@ -0,0 +1,1244 @@
+                    log.debug(f"Tolerating recreation of {would_be_created_file} in {patch_id}")
+                    os.remove(full_path)
+                else:
+                    log.warning(f"File {would_be_created_file} already exists, but patch {patch_id} would re-create it.")
+
+        # STEP 3: Apply patch using the 'patch' utility (same as serial)
+        if diff_bytes is None:
+            real_input = diff.encode("utf-8")
+        else:
+            real_input = diff_bytes
+
+        rejects_file = tempfile.mktemp()
```

Ruff (0.15.20)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)